File: app/src/main/java/com/example/tesonet/adapter/ServersAdapter.java
package com.example.tesonet.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.tesonet.R;
import com.example.tesonet.database.models.Server;

import java.util.List;

public class ServersAdapter extends RecyclerView.Adapter<ServersAdapter.ServerViewHolder> {
    private Context context;
    private List<Server> serverList;

    public ServersAdapter(Context context, List<Server> serverList) {
        this.context = context;
        this.serverList = serverList;
    }

    @NonNull
    @Override
    public ServersAdapter.ServerViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.single_listitem, parent, false);
        return new ServerViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ServersAdapter.ServerViewHolder holder, int position) {
        holder.itemView.setTag(serverList.get(position));

        Server server = serverList.get(position);

        holder.countryTextView.setText(server.getName());
        holder.distanceTextView.setText(server.getDistance() + " km");
    }

    @Override
    public int getItemCount() {
        return serverList.size();
    }

    class ServerViewHolder extends RecyclerView.ViewHolder {
        TextView countryTextView;
        TextView distanceTextView;


        public ServerViewHolder(@NonNull View itemView) {
            super(itemView);

            countryTextView = itemView.findViewById(R.id.locationTextView);
            distanceTextView = itemView.findViewById(R.id.distanceTextView);
        }
    }
}

File: Assigment/app/src/main/java/assignment/tesonet/homework/adapter/ServersAdapter.kt
package assignment.tesonet.homework.adapter

import android.content.Context
import android.databinding.DataBindingUtil
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.ViewGroup
import assignment.tesonet.homework.R
import assignment.tesonet.homework.api.response.Server
import assignment.tesonet.homework.databinding.ServerItemBinding

class ServersAdapter(private val context: Context, private val dataList : List<Server>) : RecyclerView.Adapter<ServersAdapter.ServerViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ServerViewHolder {
        val layoutInflater = LayoutInflater.from(context)
        val binding: ServerItemBinding = DataBindingUtil.inflate(layoutInflater, R.layout.server_item, parent, false)
        return ServerViewHolder(binding)
    }

    override fun onBindViewHolder(holder: ServerViewHolder, position: Int) {
        holder.bind(dataList[position])
    }

    override fun getItemCount(): Int {
        return dataList.size
    }

    class ServerViewHolder(private val binding: ServerItemBinding) : RecyclerView.ViewHolder(binding.root) {
        fun bind(server: Server) {
            binding.serverItem = server
            binding.executePendingBindings()
        }
    }
}